File: KnowledgeAccountingSystem/BLL/Models/UserSkillModel.cs
using System.ComponentModel.DataAnnotations;

namespace BLL.Models
{
    public class UserSkillModel
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "Skill is required")]
        public int? SkillId { get; set; }

        [Required(ErrorMessage = "KnowledgeLevel is required")]
        public int? KnowledgeLevelId { get; set; }
    }
}

File: KnowledgeAccountingSystem/PL/ClientApp/src/app/_models/user/userSkillModel.ts
import { KnowledgeLevelModel } from "../skill/knowledgeLevelModel";
import { SkillModel } from "../skill/skillModel";

export class UserSkillModel{
    constructor(
        public id?: number,
        public skillId?: number,
        public knowledgeLevelId?: number) { }  
}


